nifi-commons/nifi-flowfile-packager/src/main/java/org/apache/nifi/util/FlowFilePackagerV3.java

```diff
@@ -36,8 +36,8 @@ public void packageFlowFile(final InputStream in, final OutputStream out, final
         } else {
             writeFieldLength(out, attributes.size()); //write out the number of attributes
             for (final Map.Entry<String, String> entry : attributes.entrySet()) { //write out each attribute key/value pair
-                writeString(entry.getKey(), out);
-                writeString(entry.getValue(), out);
+                writeString(entry.getKey() == null ? "" : entry.getKey(), out);
```

Comment:
@mosermw how could a flowfile attribute key be null?  Do we allow that somehow?

Reply:
I don't think the NiFi framework allows null for an attribute key.  However, the Java HashMap allows a null key.  I felt that FlowFilePackagerV3 is generic enough that it could be used in unit tests where the framework isn't present to enforce that restriction.  Essentially, I changed this line to be safe in "you never know" situations.

I thought of not outputting an attribute with a null key at all, but that would throw off the number of attributes written by `writeFieldLength` on line 37.

Reply:
I see.  Yeah if that were just test code then fair enough.  But it being in the actual packager and the way it handles it makes it seem like that would be ok/handled.  We dont want to give that impression.  If you detect a null key you should reject the whole thing (the given flowfile/attrs are not valid and should not be packaged).  I recommend simply not doing this null check and fixing anything that tried to pass nulls.

Reply:
OK that makes sense.  Successfully handling a null key allows bad behavior.  Developers need to be placed in timeout if they cause a NullPointerException!

Reply:
Thanks - otherwise it seemed ready to roll.  Happy to merge to both lines.  I'll keep an eye for a commit that dumps the null thing